### ISSUE TITLE
-restrict-to-path and ./

### ISSUE THREAD

**Issue by coelner** [borgbackup/borg#5811]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes
https://borgbackup.readthedocs.io/en/stable/deployment/central-backup-server.html?highlight=restrict%20path
Maybe related to #2583 

## Is this a BUG / ISSUE report or a QUESTION?

Question, maybe a bug.

#### Your borg version (borg -V).
`client: org 1.1.16`
`server: borg 1.1.16`

#### Operating system (distribution) and version.
`client: archlinux 5.12.3-arch1-1 #1 `
`server: armbian 5.4.116-odroidxu4 #21.05.1 SMP `

#### Hardware / network configuration, and filesystems used.
net:  client <->wifi<->switch<->lan<->server
client: btrfs on ssd
server: btrfs on hdd

#### How much data is handled by borg?
~50GB 
#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`repository Location: ssh://borgbackup@server.lan/./client`

- `command="cd /mnt/backup/client; /usr/bin/borg serve --restrict-to-path /mmt/backup/client",restrict ssh-ed25519 {pubkey} user@client`
```
client: borg info borgbackup@server.lan:client
Repository path not allowed: /mnt/backup/client/client

client: borg info borgbackup@server.lan:.
Repository path not allowed: /mnt/backup/client

client: borg info borgbackup@server.lan/mnt/backup/client  # note: missing colon in scp-style url
Repository path not allowed: /mnt/backup/client

client: borg info borgbackup@server.lan/mnt/backup  # note: missing colon in scp-style url
Repository path not allowed: /mnt/backup
```

## Describe the problem you're observing.
Following the guide to limit the ssh access leads to a complete deny.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
toggle the following lines in the authorized_keys
   - `ssh-ed25519 {pubkey} user@client`
   - `#command="cd /mnt/backup/client; /usr/bin/borg serve --restrict-to-path /mmt/backup/client",restrict ssh-ed25519 {pubkey} user@client`

#### Include any warning/errors/backtraces from the system logs

the printenv for the specific borgbackup@server user under a direct terminal session:
```
$ printenv
USER=borgbackup
LANGUAGE=de_DE.UTF-8
SSH_CLIENT=192.168.0.100 43408 22
XDG_SESSION_TYPE=tty
HOME=/mnt/backup/
MOTD_SHOWN=pam
SSH_TTY=/dev/pts/2
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1004/bus
LOGNAME=borgbackup
XDG_SESSION_CLASS=user
TERM=xterm-256color
XDG_SESSION_ID=2355
PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
XDG_RUNTIME_DIR=/run/user/1004
LANG=de_DE.UTF-8
LC_MESSAGES=en_US.UTF-8
SHELL=/bin/sh
PWD=/mnt/backup
LC_ALL=de_DE.UTF-8
SSH_CONNECTION=192.168.0.100 43408 192.168.0.2 22
```
```
mount | grep mnt
/dev/sda1 on /mnt type btrfs (rw,relatime,space_cache,subvolid=5,subvol=/)
```

The behaviour get fixed by this:
  - `command="cd /mnt/backup/client; /usr/bin/borg serve --restrict-to-path .",restrict ssh-ed25519 {pubkey} user@client`
```
borg info borgbackup@server.lan:client
Repository borgbackup@server.lan:client does not exist.

borg info borgbackup@server.lan:/mnt/backup/client
Enter passphrase for key ssh://borgbackup@server.lan/mnt/backup/client: 

borg info borgbackup@server.lan:.
Enter passphrase for key ssh://borgbackup@server.lan/./.: 
```
OR better

  - `command="cd /mnt/backup; /usr/bin/borg serve --restrict-to-path client",restrict ssh-ed25519 {pubkey} user@client`
```
borg info borgbackup@server.lan:client
Enter passphrase for key ssh://borgbackup@server.lan/./client: 

borg info borgbackup@server.lan:/mnt/backup/client
Enter passphrase for key ssh://borgbackup@server.lan/mnt/backup/client:

borg info borgbackup@server.lan:.
Repository path not allowed: /mnt/backup
```
Maybe I missed something or I init the repository the wrong way.

**Comment by ThomasWaldmann** [comment]
Hmm, that looks like a bug (when used with an absolute path) ... - I'll try to reproduce.

**Comment by ThomasWaldmann** [comment]
Some of your URLs are bad, see https://borgbackup.readthedocs.io/en/stable/usage/general.html#repository-urls .

There is always a colon (`:`) after the `user@host`.

Can you fix your usage and update this issue with the results you get then?

**Comment by ThomasWaldmann** [comment]
Added another test to see what happens in case of the missing colon:
```
    def test_invalid_scp(self, monkeypatch, keys_dir):
        monkeypatch.delenv('BORG_REPO', raising=False)
        # note: invalid scp style syntax: colon missing after user@host!
>       assert repr(Location('user@host/some/path')) == \
            "Location(proto='ssh', user='user', host='host', port=None, path='/some/path', archive=None)"
E       AssertionError: assert 'Location(pro...archive=None)' == 'Location(pro...archive=None)'
E         - Location(proto='ssh', user='user', host='host', port=None, path='/some/path', archive=None)
E         ?                 ^^^        ^^^ --       ^^ ^^^
E         + Location(proto='file', user=None, host=None, port=None, path='user@host/some/path', archive=None)
E         ?                 ^^^^        ^^^        ^ ^^                   +++++++++
```

So, without the colon, it gets parsed in an unexpected way: as a relative local path `user@host/some/path`.

**Comment by ThomasWaldmann** [comment]
In general, such parsing issues can be avoided by using the `ssh://` or `file://` style of repo URLs.

**Comment by ThomasWaldmann** [comment]
So, the parsing issue explains some of your issues, but not all.

If you rerun the commands, please add `--debug` to get more log output.

**Comment by coelner** [comment]
> Some of your URLs are bad, see https://borgbackup.readthedocs.io/en/stable/usage/general.html#repository-urls .
> 
> There is always a colon (`:`) after the `user@host` **if you do not use the ssh://... form**.
> 
> Can you fix your usage and update this issue with the results you get then?

` command="cd /mnt/backup/client; /usr/bin/borg serve --restrict-to-path /mnt/backup/client",restrict ssh-ed25519 {pubkey} user@client` 
```
borg --debug info borgbackup@server.lan:/mnt/backup/
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.69 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/mnt/backup'
Remote: Resolved repository path to '/mnt/backup'
Repository path not allowed: /mnt/backup
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 155, in wrapper
    repository = RemoteRepository(location, create=create, exclusive=argument(args, exclusive),
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 591, in __init__
    do_open()
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 586, in do_open
    self.id = self.open(path=self.location.path, create=create, lock_wait=lock_wait,
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 477, in do_rpc
    return self.call(f.__name__, named, **extra)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 712, in call
    for resp in self.call_many(cmd, [args], **kw):
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 780, in call_many
    handle_error(unpacked)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 750, in handle_error
    raise PathNotAllowed(args[0].decode())
borg.remote.PathNotAllowed: Repository path not allowed: /mnt/backup

Platform: Linux client 5.12.5-arch1-1 #1 SMP PREEMPT Wed, 19 May 2021 10:32:40 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 82513  CWD: /home/user
sys.argv: ['/home/user/.local/bin/borg', '--debug', 'info', 'borgbackup@server.lan:/mnt/backup/']
SSH_ORIGINAL_COMMAND: None



borg --debug info borgbackup@server.lan:/mnt/backup/client
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.69 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/mnt/backup/client'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/mnt/backup/client: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/./.'
Warning: The repository at location ssh://borgbackup@server.lan/mnt/backup/client was previously located at ssh://borgbackup@server.lan/./.
Do you want to continue? [yN] Y
security: updating location stored in cache and security dir
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/mnt/backup/client'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/mnt/backup/client
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 252 B bytes sent, 4.30 kB bytes received, 6 messages sent


borg --debug info borgbackup@server.lan:.
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.72 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'.'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/./.: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/mnt/backup/client'
Warning: The repository at location ssh://borgbackup@server.lan/./. was previously located at ssh://borgbackup@server.lan/mnt/backup/client
Do you want to continue? [yN] Y
security: updating location stored in cache and security dir
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/./.'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/./.
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 228 B bytes sent, 4.28 kB bytes received, 6 messages sent
```

I think I struggled with the repo path, the archive name, the `:` for ssh port and the `:` to  separate it from the file path.

But 
```
borg --debug info ssh://borgbackup@server.lan/.
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.69 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/'
Remote: Resolved repository path to '/'
Repository path not allowed: /
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 155, in wrapper
    repository = RemoteRepository(location, create=create, exclusive=argument(args, exclusive),
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 591, in __init__
    do_open()
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 586, in do_open
    self.id = self.open(path=self.location.path, create=create, lock_wait=lock_wait,
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 477, in do_rpc
    return self.call(f.__name__, named, **extra)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 712, in call
    for resp in self.call_many(cmd, [args], **kw):
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 780, in call_many
    handle_error(unpacked)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 750, in handle_error
    raise PathNotAllowed(args[0].decode())
borg.remote.PathNotAllowed: Repository path not allowed: /

Platform: Linux client 5.12.5-arch1-1 #1 SMP PREEMPT Wed, 19 May 2021 10:32:40 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 83032  CWD: /home/user
sys.argv: ['/home/user/.local/bin/borg', '--debug', 'info', 'ssh://borgbackup@server.lan/.']
SSH_ORIGINAL_COMMAND: None



borg --debug info ssh://borgbackup@server.lan:22/.
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', '-p', '22', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.69 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/'
Remote: Resolved repository path to '/'
Repository path not allowed: /
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 155, in wrapper
    repository = RemoteRepository(location, create=create, exclusive=argument(args, exclusive),
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 591, in __init__
    do_open()
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 586, in do_open
    self.id = self.open(path=self.location.path, create=create, lock_wait=lock_wait,
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 477, in do_rpc
    return self.call(f.__name__, named, **extra)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 712, in call
    for resp in self.call_many(cmd, [args], **kw):
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 780, in call_many
    handle_error(unpacked)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 750, in handle_error
    raise PathNotAllowed(args[0].decode())
borg.remote.PathNotAllowed: Repository path not allowed: /

Platform: Linux client 5.12.5-arch1-1 #1 SMP PREEMPT Wed, 19 May 2021 10:32:40 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 83053  CWD: /home/user
sys.argv: ['/home/user/.local/bin/borg', '--debug', 'info', 'ssh://borgbackup@server.lan:22/.']
SSH_ORIGINAL_COMMAND: None


borg --debug info ssh://borgbackup@server.lan
usage: borg info [-h] [--critical] [--error] [--warning] [--info] [--debug] [--debug-topic TOPIC] [-p] [--log-json]
                 [--lock-wait SECONDS] [--bypass-lock] [--show-version] [--show-rc] [--umask M] [--remote-path PATH]
                 [--remote-ratelimit RATE] [--consider-part-files] [--debug-profile FILE] [--rsh RSH] [--json]
                 [-P PREFIX | -a GLOB] [--sort-by KEYS] [--first N | --last N]
                 [REPOSITORY_OR_ARCHIVE]
borg info: error: argument REPOSITORY_OR_ARCHIVE: Invalid location format: "ssh://borgbackup@server.lan"


borg --debug info ssh://borgbackup@server.lan/./
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.59 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/./'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/./: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/./.'
Warning: The repository at location ssh://borgbackup@server.lan/./ was previously located at ssh://borgbackup@server.lan/./.
Do you want to continue? [yN] y
security: updating location stored in cache and security dir
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/./'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/./
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 230 B bytes sent, 4.28 kB bytes received, 6 messages sent


borg --debug info ssh://borgbackup@server.lan/./.
using builtin fallback logging configuration
35 self tests completed in 0.16 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.68 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/./'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/./: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/./'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/./'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/./
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 230 B bytes sent, 4.28 kB bytes received, 6 messages sent



borg --debug info ssh://borgbackup@server.lan/././././.
using builtin fallback logging configuration
35 self tests completed in 0.14 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.73 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'/./'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/./: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/./'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/./'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/./
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 230 B bytes sent, 4.28 kB bytes received, 6 messages sent
```

**Comment by coelner** [comment]
I'm not sure how I initialized the repo back then. 

> Remote repositories with relative paths, alternative syntax with port:
> ssh://user@host:port/./path/to/repo - path relative to current directory

I think I overlooked the fixed suffix `/./` to declare the current path and misunderstand that this

> borg init backup@backup01.srv.local:/home/backup/repos/johndoe.clnt.local/pictures

`pictures` in this command is a folder and not a virtual placeholder for the repository name. I guess I assumed that the repo itself has got its own name and maybe fixed it up with the archive name.
```
```

**Comment by ThomasWaldmann** [comment]
I directly responded in the command output you posted, see the lines starting with `^^^` (also shorted uninteresting output lines).

Borg repos are filesystem directories, either local or remote, so yes, "repo name" and final directory name are equivalent.

I still have the impression that you missed one case from your original report that should have worked, but didn't.

**Comment by ThomasWaldmann** [comment]
```
command="cd /mnt/backup/client; /usr/bin/borg serve --restrict-to-path /mmt/backup/client",restrict ssh-ed25519 {pubkey} user@client

$ borg info --debug borgbackup@server.lan:client
Repository path not allowed: /mnt/backup/client/client

$ borg info --debug borgbackup@server.lan:.
Repository path not allowed: /mnt/backup/client
```
Can you retry these? These are expected to work.

**Comment by coelner** [comment]
sorry for the late response, I was abroad.
```
borg info --debug borgbackup@server.lan:client
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.67 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'client'
Remote: Resolved repository path to '/mnt/backup/client/client'
Repository borgbackup@server.lan:client does not exist.
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/home/user/.local/lib/python3.9/site-packages/borg/archiver.py", line 155, in wrapper
    repository = RemoteRepository(location, create=create, exclusive=argument(args, exclusive),
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 591, in __init__
    do_open()
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 586, in do_open
    self.id = self.open(path=self.location.path, create=create, lock_wait=lock_wait,
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 477, in do_rpc
    return self.call(f.__name__, named, **extra)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 712, in call
    for resp in self.call_many(cmd, [args], **kw):
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 780, in call_many
    handle_error(unpacked)
  File "/home/user/.local/lib/python3.9/site-packages/borg/remote.py", line 731, in handle_error
    raise Repository.DoesNotExist(self.location.orig)
borg.repository.Repository.DoesNotExist: Repository borgbackup@server.lan:client does not exist.

Platform: Linux client 5.12.6-arch1-1 #1 SMP PREEMPT Sun, 23 May 2021 00:45:50 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.5 msgpack: 0.5.6.+borg1
PID: 53588  CWD: /home/user
sys.argv: ['/home/user/.local/bin/borg', 'info', '--debug', 'borgbackup@server.lan:client']
SSH_ORIGINAL_COMMAND: None




borg info --debug borgbackup@server.lan:.
using builtin fallback logging configuration
35 self tests completed in 0.10 seconds
SSH command line: ['ssh', 'borgbackup@server.lan', 'borg', 'serve', '--umask=077', '--debug']
Remote: using builtin fallback logging configuration
Remote: 35 self tests completed in 0.70 seconds
Remote: using builtin fallback logging configuration
Remote: Initialized logging system for JSON-based protocol
Remote: Resolving repository path b'.'
Remote: Resolved repository path to '/mnt/backup/client'
Remote: Verified integrity of /mnt/backup/client/index.227
Enter passphrase for key ssh://borgbackup@server.lan/./.: 
TAM-verified manifest
security: read previous location 'ssh://borgbackup@server.lan/./.'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Verified integrity of /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426/chunks
security: read previous location 'ssh://borgbackup@server.lan/./.'
security: read manifest timestamp '2021-05-17T15:00:45.529212'
security: determined newest manifest timestamp as 2021-05-17T15:00:45.529212
security: repository checks ok, allowing access
Repository ID: 875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Location: ssh://borgbackup@server.lan/./.
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/user/.cache/borg/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
Security dir: /home/user/.config/borg/security/875db4b2f75ff0e91383233b85e826e01c37d0ca47fd2462ec0406019ef8e426
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              222.01 GB            154.65 GB             65.80 GB

                       Unique chunks         Total chunks
Chunk index:                  874399              2117437
RemoteRepository: 228 B bytes sent, 4.28 kB bytes received, 6 messages sent

```

**Comment by ThomasWaldmann** [comment]
OK, it now says there is no repo at `/mnt/backup/client/client` (on the repo server) - guess that is correct?

Also, it works for the repo at '/mnt/backup/client'.

So, guess the whole issue was due to the parse confusion triggered by the missing colon.

**Comment by coelner** [comment]
That is correct, I guess that was my fault due the misunderstanding that a repo is the last folder (name) and not a placeholder.

Right, and the warning with the moved path. Which is correlated by the suffix `/./` and my usage. However it is irritating in this case, because it resolve as a filesystem path to the same.

Thanks for your support

**Comment by zothar** [comment]
For anyone else that might see this later, I'll note:

The value of --restrict-to-path starts with "/mmt/" but the other paths start with the more common "/mnt/".  If the usage was as written, this may have been the result of a hard-to-see typo of "M" instead of "N" in the second letter/third character of the --restrict-to-path value.

I came across this issue when trying to figure out why I was receiving a "Repository path not allowed" before I understood that some of my scripting was sending a different repo path to a remote server than was expected by the --restrict-to-path value of the authorized SSH identity and in my case that was because the script had a bug caused by a copying of the script of one for another repository and forgetting to update the name of the SSH identity file to match the new repository, so the remote server's SSH was being authenticated with the identity of repo A but was being asked to back up to repo B.  Reading this issue helped me realize my error, but re-reading, I saw the probable typo.

At least in my case, using borg version 1.2.8, the "Repository path not allowed" error message would have been more immediately obvious if it had instead said something like "Repository path not allowed by --restrict-to-path".
